app/js/lib/view_handler/display_bank_data.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var objectAssign = require( 'object-assign' ),
+	_ = require( 'lodash' ),
+	BankDataDisplayHandler = {
+		accountNumber: null,
+		bankCode: null,
+		bankName: null,
+		update: function ( formContent ) {
+			if ( formContent.debitType === 'sepa' ) {
```

Comment:
The validator returns all the bank data (including non-sepa data when sepa data was entered and vice-versa). They could all be set here, otherwise the server-side validation will (unnecessarily) try to complement the missing data.